src/libs/TransactionUtils.ts

```diff
@@ -75,6 +79,29 @@ function isManualRequest(transaction: Transaction): boolean {
     return getRequestType(transaction) === CONST.IOU.REQUEST_TYPE.MANUAL;
 }
 
+/**
+ * Returns the type of action for the given transaction
+ */
+function getTransactionActionType(transaction: Transaction): IOUActionType {
+    if (isSplitRequest(transaction)) {
+        return CONST.IOU.TYPE.SPLIT;
+    }
+    return getRequestType(transaction);
+}
+
+/**
+ * Returns a sorted array of transactions based on the type of action
+ */
+function getTransactionsByActionType(actionType?: IOUActionType): Array<OnyxEntry<Transaction>> {
+    return (
+        Object.values(allTransactions ?? {})
+            .filter((transaction): transaction is Transaction => transaction != null && actionType === getTransactionActionType(transaction))
+            // String based sorting of dates having format [YYYY-MM-DD HH:MM:SS.mmm]
+            .sort((transactionA, transactionB) => (transactionA?.created && transactionB?.created && transactionA?.created > transactionB?.created ? 1 : -1))
+            .reverse()
```

Comment:
Why do we need to sort and then reverse? By the looks of it, we can achieve the same if we just flip the condition while sorting:

```suggestion
            .sort((transactionA, transactionB) => (transactionA?.created && transactionB?.created && transactionA?.created < transactionB?.created ? 1 : -1))
```